### ISSUE TITLE
fix(element_lang_valid) rule violations are not listed correctly in Accessibility Checker Rules

### ISSUE THREAD

**Issue by philljenkins** [IBMa/equal-access#1412]
The ruleID `element_lang_valid` is displayed in the rule list against the correct requirement but does not display the list of reasons correctly.
See [element_lang_valid.ts](https://github.com/IBMa/equal-access/blob/master/accessibility-checker-engine/src/v4/rules/element_lang_valid.ts)

All four (4) of these reasons should show as Violations against **_3.1.2 Language of Parts (AA)_** in the list of [Accessibility Checker Rules](https://unpkg.com/accessibility-checker-engine@3.1.47/help/rules.html) 
           ` "Fail_1": "Specified 'lang' attribute does not include a valid primary language",`
           ` "Fail_2": "Specified 'lang' attribute does not conform to BCP 47",`
           ` "Fail_3": "Specified 'xml:lang' attribute does not include a valid primary language",`
           ` "Fail_4": "Specified 'xml:lang' attribute does not conform to BCP 47",`
 
but none are listed against **_3.1.2 Language of Parts (AA)_**, however `html_lang_valid` **_is_** listed correctly against **_3.1.1 Language of Page (A)_** 
hint: both ruleIDs are in the same [element_lang_valid.ts](https://github.com/IBMa/equal-access/blob/master/accessibility-checker-engine/src/v4/rules/element_lang_valid.ts) file

- see screenshot:
![image](https://user-images.githubusercontent.com/16193609/234311007-b73b3e56-ef30-4201-bacd-57793da60c6f.png)

**Comment by philljenkins** [comment]
There is probably something wrong in the logic that creates the [Accessibility Checker Rules list](https://unpkg.com/accessibility-checker-engine@3.1.47/help/rules.html) that doesn't account for two ruleIDs in the same ___.ts file because the logic that defines the reasons isn't in the rule, but rather in a separate function.

**Comment by philljenkins** [comment]
@shunguoy I think this means that either we split the ruleIDs into separate ___.ts files (and add the appropriate dependencies) or fix the logic in publishing the list of the [Accessibility Checker Rules list](https://unpkg.com/accessibility-checker-engine@3.1.47/help/rules.html) by Requirement.

**Comment by philljenkins** [comment]
I also note that the single [aria_semantics.ts](https://github.com/IBMa/equal-access/blob/master/accessibility-checker-engine/src/v4/rules/aria_semantics.ts) file includes two (2) ruleIDs and yet they map correctly in the [Accessibility Checker Rules list](https://unpkg.com/accessibility-checker-engine@3.1.47/help/rules.html) to the same requirement (not two different requirements):

> **4.1.2 Name, Role, Value [A]** For all user interface components ...
> - **aria_semantics_role**: ARIA roles must be valid for the element to which they are assigned
> Violation — [The ARIA role '{0}' is not valid for the element <{1}>](https://unpkg.com/accessibility-checker-engine@3.1.47/help/en-US/aria_semantics_role.html#%7B%22message%22%3A%22The%20ARIA%20role%20'%7B0%7D'%20is%20not%20valid%20for%20the%20element%20%3C%7B1%7D%3E%22%2C%22msgArgs%22%3A%5B%22%7B0%7D%22%2C%22%7B1%7D%22%2C%22%7B3%7D%22%2C%22%7B4%7D%22%5D%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_1%22%7D)
> Violation — [The ARIA role '{0}' is not valid for the element <{1}> and may be ignored by the browser since the element is focusable](https://unpkg.com/accessibility-checker-engine@3.1.47/help/en-US/aria_semantics_role.html#%7B%22message%22%3A%22The%20ARIA%20role%20'%7B0%7D'%20is%20not%20valid%20for%20the%20element%20%3C%7B1%7D%3E%20and%20may%20be%20ignored%20by%20the%20browser%20since%20the%20element%20is%20focusable%22%2C%22msgArgs%22%3A%5B%22%7B0%7D%22%2C%22%7B1%7D%22%2C%22%7B3%7D%22%2C%22%7B4%7D%22%5D%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_2%22%7D)
> - **aria_attribute_allowed**: ARIA attributes must be valid for the element and ARIA role to which they are assigned
> Violation — [The ARIA attributes "{0}" are not valid for the element <{1}> with ARIA role "{2}"](https://unpkg.com/accessibility-checker-engine@3.1.47/help/en-US/aria_attribute_allowed.html#%7B%22message%22%3A%22The%20ARIA%20attributes%20%5C%22%7B0%7D%5C%22%20are%20not%20valid%20for%20the%20element%20%3C%7B1%7D%3E%20with%20ARIA%20role%20%5C%22%7B2%7D%5C%22%22%2C%22msgArgs%22%3A%5B%22%7B0%7D%22%2C%22%7B1%7D%22%2C%22%7B3%7D%22%2C%22%7B4%7D%22%5D%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_invalid_role_attr%22%7D)
> Violation — [The ARIA attributes "{0}" are not valid for the element <{1}> with implicit ARIA role "{2}"](https://unpkg.com/accessibility-checker-engine@3.1.47/help/en-US/aria_attribute_allowed.html#%7B%22message%22%3A%22The%20ARIA%20attributes%20%5C%22%7B0%7D%5C%22%20are%20not%20valid%20for%20the%20element%20%3C%7B1%7D%3E%20with%20implicit%20ARIA%20role%20%5C%22%7B2%7D%5C%22%22%2C%22msgArgs%22%3A%5B%22%7B0%7D%22%2C%22%7B1%7D%22%2C%22%7B3%7D%22%2C%22%7B4%7D%22%5D%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_invalid_implicit_role_attr%22%7D)

**Comment by philljenkins** [comment]
test cases: test/v2/checker/accessibility/rules/html_lang_valid_ruleunit/*
test/v2/checker/accessibility/rules/element_lang_valid_ruleunit/*

- Screenshot of the rule listing showing the reason codes now ( compare to the image above):
<img width="814" alt="image" src="https://github.com/IBMa/equal-access/assets/19577713/d5b1b43f-d0ff-456f-a072-f049b86bd4b4">

**Comment by philljenkins** [comment]
@tombrunet how did you create the screenshot above of the rule listing without deploying? 

Once deployed, I see the rule listing on the Requirements page at 
https://www.ibm.com/able/requirements/checker-rule-sets
